configure.d/conf_framework

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+#
+# Copyright(c) 2012-2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+SCRIPTPATH=`dirname $0`
+SCRIPTPATH=`realpath $SCRIPTPATH`
+MODULE_FILE=$SCRIPTPATH/test_mod.c
+OBJ_MOD=$SCRIPTPATH/test_mod.o
+KERN_VER=`uname -r`
+NPROC=`nproc`
+DEFINE_FILE=$SCRIPTPATH/../modules/generated_defines.h
+
+
+add_define() {
+   echo -n "#define " >> $DEFINE_FILE
+   for arg in "$@"; do
+       echo -e "$arg" >> $DEFINE_FILE
+   done
+   echo "" >> $DEFINE_FILE
+}
+
+add_function() {
+	for arg in "$@"; do
```

Comment:
This function can be improved significantly. 

1. Replace 'for' loop with 
printf $1 >> $DEFINE_FILE
2. At the end of framework file override internal field separator
IFS='*'
3. Now you can specify functions in plain text, like this:
add_function "
           int fun(int art)
           {
             ...
           }"